nexus/lib/qmcpack.py

```diff
@@ -384,6 +384,28 @@ def process_jastrow(wf):
             else:
                 self.error('incorporating wavefunction from '+sim.__class__.__name__+' has not been implemented')
             #end if
+        elif result_name=='gc_occupation':
+            from qmcpack_converters import gcta_occupation
+            if not isinstance(sim,Pw2qmcpack):
+                msg = 'grand-canonical occupation require Pw2qmcpack'
+                self.error(msg)
+            #endif
+            # step 1: extract Fermi energy for each spin from nscf
+            if len(sim.dependencies) != 1:
+                msg = 'need scf/nscf calculation for Fermi energy'
+                self.error(msg)
+            #end if
+            nscf = sim.dependencies[1].sim
+            na = nscf.load_analyzer_image()
+            Ef_list = na.fermi_energies
+            # step 2: analyze ESH5 file for states below Fermi energy
+            pa = sim.load_analyzer_image()
+            pa.analyze(Ef_list=Ef_list)
```

Comment:
This step might get repeated many times and get heavy (multiple calculations depending on Pw2qmcpack for gcta).  Please check first for presence of 'wfh5' in pa, then proceed with analyze and save if not present.

if 'wfh5' not in pa:
    pa.analyze(Ef_list=Ef_list)
    sim.save_analyzer_image(pa)
#end if

with `save_analyzer_image` added to class `Simulation` as

def save_analyzer_image(self,analyzer):
    analyzer.save(os.path.join(self.imresdir,self.analyzer_image))
#end def save_analyzer_image

Reply:
yes, good point.
will cache `pa` in next push